ui/src/app/kvm/views/KVMDetails/KVMConfiguration/KVMConfiguration.tsx

```diff
@@ -1,28 +1,132 @@
 import { useDispatch, useSelector } from "react-redux";
 import { useParams } from "react-router";
+import { useHistory } from "react-router-dom";
+import * as Yup from "yup";
 
-import { pod as podActions } from "app/base/actions";
-import { pod as podSelectors } from "app/base/selectors";
+import type { RootState } from "app/store/root/types";
+import type { Pod } from "app/store/pod/types";
+import {
+  pod as podActions,
+  resourcepool as resourcePoolActions,
+  tag as tagActions,
+  zone as zoneActions,
+} from "app/base/actions";
+import {
+  pod as podSelectors,
+  resourcepool as resourcePoolSelectors,
+  tag as tagSelectors,
+  zone as zoneSelectors,
+} from "app/base/selectors";
 import { useWindowTitle } from "app/base/hooks";
+import { formatErrors } from "app/utils";
+import FormCard from "app/base/components/FormCard";
+import FormCardButtons from "app/base/components/FormCardButtons";
+import FormikForm from "app/base/components/FormikForm";
+import KVMConfigurationFields from "./KVMConfigurationFields";
 
-import type { RootState } from "app/store/root/types";
+const KVMConfigurationSchema = Yup.object().shape({
+  cpu_over_commit_ratio: Yup.number().required("CPU overcommit ratio required"),
+  memory_over_commit_ratio: Yup.number().required(
+    "Memory overcommit ratio required"
+  ),
+  pool: Yup.string().required("Resource pool required"),
+  power_address: Yup.string().required("Address required"),
+  password: Yup.string(),
+  tags: Yup.array().of(Yup.string()),
+  type: Yup.string().required("Type required"),
+  zone: Yup.string().required("Zone required"),
+});
+
+export type KVMConfigurationValues = {
+  cpu_over_commit_ratio: Pod["cpu_over_commit_ratio"];
+  memory_over_commit_ratio: Pod["memory_over_commit_ratio"];
+  pool: Pod["pool"];
+  power_address: Pod["power_address"];
+  password: Pod["password"] | Pod["power_pass"];
+  tags: Pod["tags"];
+  type: Pod["type"];
+  zone: Pod["zone"];
+};
 
 const KVMConfiguration = (): JSX.Element => {
   const dispatch = useDispatch();
+  const history = useHistory();
   const { id } = useParams();
-
   const pod = useSelector((state: RootState) =>
     podSelectors.getById(state, Number(id))
   );
+  const podErrors = useSelector(podSelectors.errors);
+  const podSaved = useSelector(podSelectors.saved);
+  const podSaving = useSelector(podSelectors.saving);
+  const resourcePoolsLoaded = useSelector(resourcePoolSelectors.loaded);
+  const tagsLoaded = useSelector(tagSelectors.loaded);
+  const zonesLoaded = useSelector(zoneSelectors.loaded);
+
+  const errors = formatErrors(podErrors);
 
   useWindowTitle(`KVM ${`${pod?.name} ` || ""} configuration`);
 
   useEffect(() => {
     dispatch(podActions.fetch());
+    dispatch(resourcePoolActions.fetch());
+    dispatch(tagActions.fetch());
+    dispatch(zoneActions.fetch());
   }, [dispatch]);
 
-  return <></>;
+  const loaded = resourcePoolsLoaded && tagsLoaded && zonesLoaded;
+
+  if (typeof pod === "object" && loaded) {
```

Comment:
I couldn't figure out why `typeof` is used here. How come a truthy check isn't enough (also `typeof null === "object"` returns true)? I'm probably missing something obvious...

Reply:
Whatever this state is I think it'd be good to add a test for it.

Reply:
Yeah this definitely looks strange, but it's a strict mode fix. Firstly I had to fix having a non-boolean in a boolean check (since `pod` is not a boolean). But also I needed to fix betterer warnings in the conditional code block about `pod.attribute` not existing if `pod` is not an object. And I think adding `pod?.type`, `pod?.password`, etc everywhere is a bit messy.

There's has to be a better way of satisfying strict mode because I agree it's not really clear here.

Reply:
Opting instead for `if (!!pod && loaded) {`